util/rowDecoder/decoder_test.go

```diff
@@ -202,18 +194,27 @@ func (s *testDecoderSuite) TestClusterIndexRowDecoder(c *C) {
 	rd := rowcodec.Encoder{Enable: true}
 	for _, row := range testRows {
 		bs, err := tablecodec.EncodeRow(sc, row.input, row.cols, nil, nil, &rd)
-		c.Assert(err, IsNil)
-		c.Assert(bs, NotNil)
+		require.Nil(t, err)
+		require.NotNil(t, bs)
 
-		r, err := de.DecodeAndEvalRowWithMap(ctx, testutil.MustNewCommonHandle(c, 100, "abc"), bs, time.UTC, timeZoneIn8, nil)
-		c.Assert(err, IsNil)
+		r, err := de.DecodeAndEvalRowWithMap(ctx, MustNewCommonHandle(t, 100, "abc"), bs, time.UTC, timeZoneIn8, nil)
+		require.Nil(t, err)
 
 		for i, col := range cols {
 			v, ok := r[col.ID]
-			c.Assert(ok, IsTrue)
+			require.True(t, ok)
 			equal, err1 := v.CompareDatum(sc, &row.output[i])
-			c.Assert(err1, IsNil)
-			c.Assert(equal, Equals, 0)
+			require.Nil(t, err1)
+			require.Equal(t, 0, equal)
 		}
 	}
 }
+
+// TODO: local copy of testutil.MustNewCommonHandle，should use function in testutil after it migrate to testify
+func MustNewCommonHandle(t *testing.T, values ...interface{}) kv.Handle {
```

Comment:
You can just create a file named, for example, "handle.go" under "testkit" and do the todo a favor.

Reply:
sure, a little busy now hope finish before bed

Reply:
Thanks ❤️

Reply:
sorry, i met some problems and not sure how to resolve.
1. decoder_test.go unit test failed for goleak
```shell
goleak: Errors on successful test run: found unexpected goroutines:
[Goroutine 21 in state select, with go.opencensus.io/stats/view.(*worker).start on top of the stack:
goroutine 21 [select]:
go.opencensus.io/stats/view.(*worker).start(0xc000162410)
	/home/ming/go/pkg/mod/go.opencensus.io@v0.22.3/stats/view/worker.go:154 +0xcd
created by go.opencensus.io/stats/view.init.0
	/home/ming/go/pkg/mod/go.opencensus.io@v0.22.3/stats/view/worker.go:32 +0x57
```

2. key_test.go and keydecoder_test.go have cycle import with util/testkit/handle.go
```shell
# github.com/pingcap/tidb/kv
FAIL	github.com/pingcap/tidb/kv [setup failed]
package github.com/pingcap/tidb/kv (test)
	imports github.com/pingcap/tidb/util/testkit
	imports github.com/pingcap/tidb/domain
FAIL
	imports github.com/pingcap/tidb/bindinfo
	imports github.com/pingcap/tidb/sessionctx
	imports github.com/pingcap/tidb/kv: import cycle not allowed in test

```

Reply:
i need some dig to resolve the 2 problems, any suggestion is appreciated

Reply:
For goleak, it is a known goroutine that doesn't exit, you can ignore by

```go
        opts := []goleak.Option{
		goleak.IgnoreTopFunction("go.etcd.io/etcd/pkg/logutil.(*MergeLogger).outputLoop"),
		goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"),
	}
	goleak.VerifyTestMain(m, opts...)
```

for cycle import, you can try to change `package keydecoder` to `package keydecoder_test`

Reply:
thanks, helps a lot

Reply:
updated, please review again, thanks